sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1069,10 +1069,11 @@ ProgramManager::getDeviceImage(const std::string &KernelName,
                         PI_ERROR_INVALID_KERNEL_NAME);
 }
 
-RTDeviceBinaryImage &ProgramManager::getDeviceImage(
-    const std::unordered_set<RTDeviceBinaryImage *> &ImageSet,
-    const context &Context, const device &Device,
-    bool JITCompilationIsRequired) {
+template <template <typename, typename...> class Container>
```

Comment:
If I understand correctly, here we are trying to generalize this function to receive any kind of container. However, some containers might not have `push_back()`. If we want to use particularly `std::vector`, why do we need this? Otherwise, if we indeed want to keep the ability to receive any kind of container, using `push_back()` might be problematic if someone calls the function using a container not supporting `push_back()`.

Reply:
yes, I am trying to generalize it for std::vector and std::set. push_back is not used in this method. This method just iterates over container. Although I could not use iterators passed as function parameters because I need lock inside (before I took iterators).